code/datums/beam.dm

```diff
@@ -137,6 +141,39 @@
 		CHECK_TICK
 	afterDraw()
 
+/datum/beam/proc/get_x_translation_vector()
+	return (32*target.x+target.pixel_x)-(32*origin.x+origin.pixel_x)
```

Comment:
Not super important, but you could replace these 32s with world.icon_size, here and elsewhere.